source/utilities.cc

```diff
@@ -260,10 +287,18 @@ namespace aspect
                                                        n_fields,
                                                        property_name);
 
-          if (subentry_structure)
+          if (store_structure)
+            {
+              for (unsigned int i=0; i<n_fields; ++i)
+                (*n_values_per_key)[i] = 1;
+            }
+
+          if (check_structure)
             {
               for (unsigned int i=0; i<n_fields; ++i)
-                (*subentry_structure)[i] = 1;
+                AssertThrow((*n_values_per_key)[i] == 1,
+                            ExcMessage("The input parameter " + property_name + " does not have "
+                                       + "the expected number of values for field index " + std::to_string(i) + "."));
```

Comment:
I find this error message a bit confusing. If this error is thrown, it means someone specified a list of doubles in the input file with one value per key, but some other parameter was specified with multiple values per key, provided in the key:value format (or there is some other reason why the function expects more than one value per key). 

So to help the user a bit more, I would tell the user how many entries were expected, and tell them something like: "To specify more than one value per field for this input parameter, you need to use the format <key>:value1|value2|..."